### DISCUSSION TITLE
Using variables in events: state node?

### DISCUSSION THREAD

**Discussion by edge360** [zachowj/node-red-contrib-home-assistant-websocket#1184] [Q&A]
Not sure if this is supported, but it isn't working the way I expected it might. Attached is what I'm hoping to do, take a variable I have defined in global context and fetch the state when changed. Does this get evaluated using any of the entity definition settings?

Using it as {{global.some_entity}} currently, doesn't seem to do anything.

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/2661427/76639268-66db-4942-9314-331d7087bc7c)

**Comment by zachowj** [comment]
That should work as is, it takes the value when deploying this node. If the value of the global context changes after the node is deployed it will not update the trigger of the node.

**Reply by edge360** [reply]
I appreciate the response, thank you for confirming. If I try running this by manually specifying the entity, it runs. When I run it using my context, nothing ever activates which is part of the reason I'm unsure if I'm doing this right since I don't have any debug info.  State node is same format as original post, just using {{global.basement_leak_sensor}}. Using curly braces in the Input node outputs the correct entity from the context. 

I've just updated as well to latest ha/node-red/ws-plugin and am running into the same thing. 

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/2661427/0c7e891d-70b9-40aa-8bf4-2a7f6706d899)

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/2661427/8738f10d-b6d7-48b7-950d-da30a14942e9)

**Reply by zachowj** [reply]
Sorry, I was mistaken I was thinking about using environment variables, not mustache templates. 

https://nodered.org/docs/user-guide/environment-variables

**Reply by edge360** [reply]
Gotcha, so the node it can't evaluate global context within the mustache templates, but it can with ENVs. Would it evaluate any type of JSONATA with any of the entity filters?

Say I have env.LEAK_SENSOR, which format would the state node accept? Whichever I seem to just fail silently as the node state just says 'running'. 

{{env.NR_NODE_NAME}}
${env.NR_NODE_NAME}#

**Reply by zachowj** [reply]
> Would it evaluate any type of JSONATA with any of the entity filters?

no

> Say I have env.LEAK_SENSOR, which format would the state node accept? Whichever I seem to just fail silently as the node state just says 'running'.

${NR_NODE_NAME}

**Reply by chrisdrackett** [reply]
is it possible to add to this? I'm trying to do something like `sensor.temperature_${NAME}` but it does not seem to be working. I've tried other forms as well, but without any luck
